multicluster/controllers/multicluster/member/clusterset_controller.go

```diff
@@ -348,10 +363,21 @@ func (r *MemberClusterSetReconciler) updateStatus() {
 	err := r.Get(context.TODO(), namespacedName, clusterSet)
 	if err != nil {
 		klog.ErrorS(err, "Failed to read ClusterSet", "name", namespacedName)
+		return
 	}
 	status.Conditions = clusterSet.Status.Conditions
 	if (len(clusterSet.Status.Conditions) == 1 && clusterSet.Status.Conditions[0].Status != overallCondition.Status) ||
 		len(clusterSet.Status.Conditions) == 0 {
+		if overallCondition.Status == v1.ConditionTrue {
+			// Try to clean up all stale auto-generated resources in this member cluster and
+			// its corresponding stale ResourceExports in the leader cluster before updating
+			// the cluster status as ready.
+			err := r.CleanUp(context.Background())
```

Comment:
It is really hard to track if we also do cleanup in `updateStatus()`. Why we need to do it here?

Imported resources should be handled by stale controller already. For ResourceExports, leader cluster should have a mechanism to guarantee cleanup is done first, before changing ClusterSet status?

Reply:
Hi @jianjuns I refined the codes and updated the PR summary, the old stale controller are split into two stale controllers in leader and member folder. Please check the details change in the PR summary.